packages/@react-stately/virtualizer/src/Virtualizer.ts

```diff
@@ -633,6 +643,30 @@ export class Virtualizer<T extends object, V, W> {
       map.set(layoutInfo.key, layoutInfo);
     }
 
+    // Views that are outside of the viewable rectangle that we don't
+    // want to be reused by Virtualizer.
+    for (let persistKey of this._persistedKeys) {
+      let key = persistKey;
+
+      while (key) {
+        let layoutInfo = this.layout.getLayoutInfo(key);
+
+        if (layoutInfo) {
+          if (copy) {
+            layoutInfo = layoutInfo.copy();
+          }
+
+          map.set(layoutInfo.key, layoutInfo);
```

Comment:
One problem: adding the persisted keys at the end means the elements get out of order in the DOM. You can see this by scrolling the focused item out of view, and seeing that it is now at the end of the list. We previously relied on the layout giving us the views in the correct order. We will have to think about how to solve this...

Reply:
I'll move this. I'm not seeing this so far. I realized I've only tested selecting a top item/row/cell and scrolling down. The focus ring shows it staying above the table/list.

Reply:
I'm not sure this is needed. I was checking how the code works and it looks like there is code to make sure the correct dom order is maintained via `_correctItemOrder()` https://github.com/adobe/react-spectrum/blob/8cd847d24ac7b1e0f778baf28bba7f14bfad89d7/packages/%40react-stately/virtualizer/src/Virtualizer.ts#L537

I did more testing and I can select an item anywhere in the list and scroll it out of view at the top or bottom of the list. It always exists in the correct relative location.

Reply:
"Luke Skywalker" is the first row in the async loading ListView example. If I focus it and then scroll out of view, I see it at the end of the list:

<img width="831" alt="image" src="https://user-images.githubusercontent.com/19409/176936000-112781c4-ef37-43e2-a10b-24dbbb573d49.png">